lib/extensions/ar_migration.rb

```diff
@@ -0,0 +1,53 @@
+module MigrationSyncHelper
+  def self.migrations_column_present?
+    ActiveRecord::Base.connection.columns("miq_regions").detect { |c| c.name == "migrations_ran" }
+  end
+
+  class HelperARClass < ActiveRecord::Base; end
+  def self.restart_subscription(s)
+    c = HelperARClass.establish_connection.connection
+    c.pglogical.subscription_disable(s.id)
+    c.pglogical.subscription_enable(s.id)
+  ensure
+    HelperARClass.remove_connection
+  end
+
+  def self.wait_for_remote_region_migration(subscription, version, wait_time = 1)
+    return unless MigrationSyncHelper.migrations_column_present?
+    region = MiqRegion.find_by(:region => subscription.provider_region)
+    until region.migrations_ran&.include?(version)
+      restart_subscription(subscription)
+      sleep(wait_time)
+      region.reload
+    end
+  end
+
+  def self.update_local_migrations_ran(version, direction)
+    return unless MigrationSyncHelper.migrations_column_present?
+    return unless (region = MiqRegion.my_region)
+
+    new_migrations = ActiveRecord::SchemaMigration.normalized_versions
+    new_migrations << version if direction == :up
+    migrations_value = ActiveRecord::Base.connection.quote(PG::TextEncoder::Array.new.encode(new_migrations))
+
+    ActiveRecord::Base.connection.exec_query(<<~SQL)
```

Comment:
Minor, but is it better to use `.update` instead of `.exec_query`?

Reply:
No? It doesn't look that way based on the implementation [[ref]](https://github.com/rails/rails/blob/v5.0.7/activerecord/lib/active_record/connection_adapters/abstract/database_statements.rb#L110-L134), but I don't really have a preference.